functions/validators.ts

```diff
@@ -0,0 +1,43 @@
+import { CustomRequest, POSTRequestBody, ValidationResult } from "./types";
+import { Octokit } from "@octokit/rest";
+
+export async function validatePOST(request: CustomRequest): Promise<ValidationResult> {
+  try {
+    const jsonData: POSTRequestBody = await request.json();
+
+    const { authToken, referralCode } = jsonData;
+
+    if (!authToken || !referralCode) {
+      console.error("Missing authToken or referralCode");
+      return { isValid: false };
+    }
+
+    const providerToken = authToken.provider_token;
+
+    if (!providerToken) {
+      console.error("Missing provider token");
+      return { isValid: false };
+    }
+
+    const octokit = new Octokit({ auth: providerToken });
+
+    try {
+      await octokit.request("GET /user");
+
+      const githubUserId = authToken.user?.user_metadata?.provider_id;
```

Comment:
You should send Github auth token instead of the whole Supabase auth object which can be manipulated with and you can get the user ID from the octokit request.
Also why are there 2 try/catch blocks?

Reply:
I was using the first one to handle the json parsing, but I will remove it